tests/README.md

```diff
@@ -1,14 +1,73 @@
+# Running and creating tests                        {#running-and-creating-tests}
 
 There are a number of tests included in RIOT. They are located in the
 [tests folder](https://github.com/RIOT-OS/RIOT/tree/master/tests). These tests
 allow basic functionality to be verified as well as provide an example of
 usage.
 
+# Running automated tests
 
-Implementing automated tests
-----------------------------
+Some tests can be performed automatically. The test automation scripts are
+defined in the `<test_application>/tests/` folder. They are written in python
+and interact through the uart with the test application code running on a
+board to do the validation. It is recommended to flash the board with the
+test just before running it because some platforms cannot be reset while
+testing.
+
+## Running single test
+
+From the test application directory run:
+
+    BOARD=<board_of_your_choice> make flash test
+
+
+An automated way of knowing if a test is available is to execute the
+'test/available' target from the test application directory.
+It executes without error if tests run by 'make test' are present.
+
+    make test/available
+
+## Running all test for particular board
+
+If you would like execute all tests for given board, you could use dedicated
+script `compile_and_test_for_board.py`
+
+Go to main RIOT directory and execute command:
+
+    ./dist/tools/compile_and_test_for_board/compile_and_test_for_board.py . <board_of_your_choice> --with-test-only --jobs=4
+
+More details concerning other available parameters provided by this tool can be found in
+[README.md file](https://github.com/RIOT-OS/RIOT/tree/master/dist/tools/compile_and_test_for_board)
+and directly in [compile_and_test_for_board.py](https://github.com/RIOT-OS/RIOT/tree/master/dist/tools/compile_and_test_for_board/compile_and_test_for_board.py) script.
+
+## Running tests that require a preliminary manual configuration
+
+Some tests need active monitoring or manual setup steps but still have some
+automated scripts. The test automation scripts are defined in the
+`<test_application>/tests-with-config/` folder.
+For running them, follow the setup or analysis documentation and use the
+`test-with-config` target.
+
+## Running tests that require root privileges
+
+Some tests require root privileges to launch their automated script. In this
+case, the test automation scripts are defined in the
+`<test_application>/tests-as-root/` folder.
+For running them, follow the setup or analysis documentation and use the
+`test-as-root` target.
+
+## Cleaning intermediate files
+
+After test execution intermediate files are not automatically deleted.
+Execution of multiple tests, especially all for particular board could generate
+many files. For example, after execution of all test for stm32f469i-disco board
+(more than 230 tests) around 7.5 GB of intermediate files are created.
+
+To delete all of this files go to main RIOT directory and execute command:
+
+    make distclean
```

Comment:
Could you hint me other 'less aggressive cleanup' methods? `distclean` was the only one which I found when after running tests for three boards my file system was completely full.

Reply:
`make clean` "Cleaning all build products for the current board" If you know know which board you just tested.
This will spare you the pkg-download for the next board you test and the doc rebuild (if you like to have a local doc)

`compile_and_test_for_board.py ` also has a `--clean-after` flag that might fit your need
or `--compile-targets clean` if you like to cleanup later

I put me this Makefile into `tests/` and `examples/` to do test-builds. (just building not testing) (the ignore list are for build chains that i do not have installed, the DONE files generated are for speed up repeated builds) not sure if this helps your Usecase.

``` Makefile 
DIRS := $(wildcard */)

IGNORE_EXAMPLES := mallinfo \
                  rust-gcoap \
                  rust-hello-world \
                  suit_update \
                  #

IGNORE_TESTS := cpu_avr8_xmega_drivers \
                cpu_efm32_features \
                cpu_efm32_drivers \
                driver_ws281x \
                driver_esp_eth \
                mcuboot \
                pkg_emlearn \
                pkg_nanopb \
                pkg_tensorflow-lite \
                rust_minimal \
                suit_manifest \
                sys_stdio_semihosting \
                ztimer_until \
                #

IGNORE := $(INGNORE) $(IGNORE_EXAMPLES) $(IGNORE_TESTS)

DIRS := $(filter-out $(addsuffix /,$(IGNORE)), $(DIRS))

CLEANLIST := $(addsuffix /bin,$(DIRS))
DIRSDONE := $(addsuffix /bin/DONE,$(DIRS))

all: $(DIRSDONE)
	@echo FINISHED ALL

$(DIRSDONE):
	$(MAKE) -C $(subst /bin/DONE,,$@)
	touch $@

$(DIRS): FORCE
	$(MAKE) -C $@

clean_all: undone
	rm -rf $(CLEANLIST)

undone:  FORCE
	rm $(DIRSDONE)

FORCE:
```

Reply:
Thanks for hints. I will test it and update this section accordingly.

Reply:
I update section concerning various methods for cleaning intermediate files. @kfessel could you review my changes?